src/services/test-helpers/mock/mockApi.ts

```diff
@@ -276,6 +289,187 @@ const referendumInfoOfAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<FundInfo>', funds);
+	});
+
+const fundsKeys = () =>
+	Promise.resolve().then(() => {
+		return [paraId1, paraId2];
+	});
+
+const parasLifecyclesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, paraLifecycleOne],
+			[paraId2, paraLifecycleTwo],
+		];
+	});
+
+const parasGenesisArgsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('ParaGenesisArgs', { parachain: true });
+	});
+
+const upcomingParasGenesisAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<ParaGenesisArgs>', {
+			parachain: true,
+		});
+	});
+
+const parasLifecyclesAt = () =>
+	Promise.resolve().then(() => {
+		return typeFactory.optionOf(paraLifecycleOne);
+	});
+
+/**
+ * Used for parachain leases
+ */
+const leasesTupleOne = typeFactory.tupleOf(
+	[accountIdOne, balanceOfOne],
+	['AccountId', 'BalanceOf']
+);
+const leasesTupleTwo = typeFactory.tupleOf(
+	[accountIdTwo, balanceOfTwo],
+	['AccountId', 'BalanceOf']
+);
+const parasOptionsOne = typeFactory.optionOf(leasesTupleOne);
+const parasOptionsTwo = typeFactory.optionOf(leasesTupleTwo);
+const vectorLeases = typeFactory.vecOf([parasOptionsOne, parasOptionsTwo]);
+
+export const slotsLeasesAt = (): Promise<Vec<Option<Tuple>>> =>
+	Promise.resolve().then(() => {
+		return vectorLeases;
+	});
+
+const slotsLeasesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, vectorLeases],
+			[paraId2, vectorLeases],
+		];
+	});
+
+/**
+ * Used for parachain Auctions
+ */
+const beingEnd = new BN(1000) as AbstractInt;
+const leasePeriodIndex = new BN(39) as AbstractInt;
+const vectorAuctions = typeFactory.vecOf([beingEnd, leasePeriodIndex]);
+
+export const auctionsInfoAt = (): Promise<Option<Vec<AbstractInt>>> =>
+	Promise.resolve().then(() => {
+		const optionAuctions = typeFactory.optionOf(vectorAuctions);
+
+		return optionAuctions;
+	});
+
+export const nullAuctionsInfoAt = (): Promise<Vec<AbstractInt>> =>
+	Promise.resolve().then(() => {
+		return vectorAuctions;
+	});
+
+const auctionCounterAt = () =>
+	Promise.resolve().then(() => {
+		const counter = new BN(4) as AbstractInt;
+
+		return counter;
+	});
+
+const auctionsWinningsAt = () =>
+	Promise.resolve().then(() => {
+		const paraId1 = rococoRegistry.createType('ParaId', 199);
+		const paraId2 = rococoRegistry.createType('ParaId', 200);
+		const tupleOne = typeFactory.tupleOf(
+			[accountIdOne, paraId1, balanceOfOne],
+			['AccountId', 'ParaId', 'BalanceOf']
+		);
+		const tupleTwo = typeFactory.tupleOf(
+			[accountIdTwo, paraId2, balanceOfTwo],
+			['AccountId', 'ParaId', 'BalanceOf']
+		);
+		const parasOptionsOne = typeFactory.optionOf(tupleOne);
+		const parasOptionsTwo = typeFactory.optionOf(tupleTwo);
+
+		const vectorWinnings = typeFactory.vecOf([
+			parasOptionsOne,
+			parasOptionsTwo,
+		]);
+		const optionWinnings = typeFactory.optionOf(vectorWinnings);
```

Comment:
Should this be longer? like a length of 7 (Or whatever the `SLOT_RANGE_COUNT` is)? The rest of the values can just be `Option::None`, but I am concerned there could be behavior that relies on the length of `auctions::Winning`. 

It is defined as `[Option<(<T as frame_system::Config>::AccountId, ParaId, BalanceOf<T>)>; SlotRange::SLOT_RANGE_COUNT]` where `SLOT_RANGE_COUNT` is the length it will always be.

Reply:
So I couldnt find where it says the minimum `SLOT_RANGE_COUNT` is, but I know it's suppose to be a maximum of 10 per the older code about a week ago.

```
/// Total number of possible sub ranges of slots.
pub const SLOT_RANGE_COUNT: usize = 10;
```
This was the comment that they had for slot range. 

Now the codes changed a bit which includes a `generate_slot_range` macro. 

`/// * A constant `SLOT_RANGE_COUNT` will count the total number of enum variants.`

Reply:
Lets hardcode it to 10 (I think we covered the rest in offline discussion.)